src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -1367,6 +1369,44 @@ def create_rotation_matrices(self):
                                              numpy.radians(float(self.txtSampleRoll.text()))])
         return env, sample*env
 
+    def check_large_calculation(self, num_points):
+        """
+        Generate a popup warning the user about a large calculation
+        :return True if the calculation should continue, False otherwise.
+        """
+
+        if self.warned_user_large_calc or num_points < 50_000: return True
+        # timings for an i7-1165G7 4-core CPU from 2020:
+        # 50k: 1.8s | 100k: 6.3s | 200k: 28.7s | 300k: 65.6s | 400k: 117.2s | 500k: 181.2s
+        # multiply by 2 & round to get a conservative estimate
+        thresholds = [
+            (100_000,      50_000,  "30 seconds"),
+            (200_000,      100_000, "a minute"),
+            (300_000,      200_000, "a couple of minutes"),
+            (400_000,      300_000, "around 5 minutes"),
+            (500_000,      400_000, "around 10 minutes"),
+            (float('inf'), 500_000, "more than an hour")
+        ]
+
+        # find the appropriate threshold and time estimate
+        for limit, exceeded_threshold, time_estimate in thresholds:
+            if num_points < limit:
+                msg = (f"The number of atoms exceeds {exceeded_threshold:,}.\n"
+                       f"This calculation may take {time_estimate} to complete. "
+                       "Do you wish to proceed?")
+                break
+
+        flag_continue = False
+        popup = QMessageBox(standardButtons = QMessageBox.Ok | QMessageBox.Cancel)
+        popup.setIcon(QMessageBox.Warning)
+        popup.setText(msg)
+        popup.setWindowTitle("Large input structure")
+        popup.setDefaultButton(QMessageBox.Cancel)
+        result = popup.exec_()
```

Comment:
I am not an expert on Qt but exec_() seems to becoming obsolete.  exec() using setters is probably the safest across PySide/PyQt flavors and future versions